docs/README.md

```diff
@@ -0,0 +1,423 @@
+```
+
+If the current active version of the secret still matches what the client
+requested, the server will report 304 Not Modified, indicating to the client
+that it still (already) has the active version. This allows the client to check
+for an update without sending secret values over the wire except when needed,
+and does not trigger an auditable access record unless the value or the
+permissions have changed.
+
+The Go client library provides a [`setec.Store`][setecstore] type that handles
+automatic updates using this polling mechanism, but the same logic can be
+implemented in any language. In Go, this looks like:
+
+```go
+import "github.com/tailscale/setec/client/setec"
+
+func main() {
+   // Construct a setec.Store that tracks a set of secrets this program needs.
+   // This will block until all the requested secrets are available.
+   // Thereafter, in the background, it will poll for new versions at
+   // approximately the given interval.
+   st, err := setec.NewStore(context.Background(), setec.StoreConfig{
+      Client: setec.Client{Server: "https://secrets.example.ts.net"},
+      Secrets: []string{
+         "prod/myprogram/secret-1",
+         "prod/myprogram/secret-2",
+      },
+      PollInterval: 24*time.Hour,
+   })
+   if err != nil {
+      log.Fatalf("NewStore: %v", err)
+   }
+
+   // Fetch a secret from the store. A setec.Secret is a handle that always
+   // delivers the current value, automatically updated as new versions become
+   // available from the server.
+   apiKey := st.Secret("prod/myprogram/secret-1")
+
+   cli := someservice.NewClient("username", apiKey.Get())
+   // ...
+}
+```
+
+Alternatively, you can obtain a [`setec.Watcher`][setecwatcher], which combines
+a secret with a channel that notifies when a new secret value is available:
+
+```go
+// Get the handle, as before.
+apiKey := st.Watcher("prod/my-program/secret-1")
+
+// Create a client with the current value.
+cli := someservice.NewClient("username", apiKey.Get())
+
+// Make a helper that will refresh the client when the secret updates.
+// This example assumes no concurrency; you may need a lock if multiple
+// goroutines will request a client at the same time.
+getClient := func() *someservice.Client {
+   select {
+   case <-apiKey.Ready():
+     cli = someservice.NewClient("username", apiKey.Get())
+   default:
+   }
+   return cli
+}
+
+// Make API calls using the helper.
+rsp, err := getClient().Method(ctx, args)
+// ...
+```
+
+### Bootstrapping and Availability
+
+A reasonable concern when fetching secrets from a network service is what
+happens if the secrets service is not reachable when a program needs to fetch a
+secret (e.g., when it is starting up). A good answer depends on the nature of
+the program: Batch processing tools can usually afford to wait and retry until
+the service becomes available. Interactive services, by contrast, may not be
+able to tolerate waiting.
+
```

Comment:
May be worth adding a paragraph here to clarify the semantics of server unavailability at runtime: if the client has initialized itself and/or uses a store as above, then it should be mostly immune to the outage as long as it does not itself restart and need to bootstrap again (in which case, this section is your friend)

Reply:
Good call. I have made an attempt at this, PTAL?